test/inductor/test_snode_runtime.py

```diff
@@ -165,6 +172,132 @@ def f(x):
         self.assertNotZero(calculate_runtime(f, *inp))
 
 
+class TestCommAnalysis(TestCase):
+    WORLD_SIZE: int = 8
+    RANKS = list(range(8))
+
+    def _verify_runtime_estimation(self, fn, inps):
+        store = FakeStore()
+        dist.init_process_group(
+            backend="fake", rank=0, world_size=self.WORLD_SIZE, store=store
+        )
+        try:
+            metrics.reset()
+            torch._dynamo.optimize(count_bytes_inductor)(fn)(*inps)
+            found_collective = False
+            for snode, runtime in metrics.node_runtimes:
```

Comment:
cool, is this an estimate from the scheduler for every node, (iirc there were some lo-fi estimates like eltwise was cheap and gemm is expensive, but I didn't remember if it was calibrated (is the collective estimator predicting actual time intervals?)

Reply:
nvm see it below. time in ns.